src/System.Runtime.Extensions/src/System/IO/PathHelper.Windows.cs

```diff
@@ -160,78 +158,34 @@ internal string Normalize(string path, bool fullCheck, bool expandShortPaths)
             }
             finally
             {
-                if (fullPath != null && fullPath.Capacity <= MaxBuilderSize)
-                {
-                    // Stash for reuse
-                    t_cachedOutputBuffer = fullPath;
-                }
+                // Clear the buffer and stash it
+                fullPath.Free();
+                s_fullPathBuffer = fullPath;
```

Comment:
I went with (2). Size doesn't matter here as I'm freeing the backing buffer to the pool- this is only to avoid the instance allocation cost.